apps/desktop/desktop/src/main/java/bisq/desktop/main/content/authorized_role/moderator/BannedUserProfileTable.java

```diff
@@ -260,12 +346,49 @@ private static class ListItem {
 
             private final UserProfile userProfile;
             private final String userName;
+            private final String reporterUserProfileId;
+            private final String reporterUserName;
+            private final Optional<UserProfile> reporterUserProfile;
+            private final String banReason;
+
+            public static ListItem from(BannedUserProfileData data,
+                                        UserProfileService userProfileService,
+                                        ModeratorService moderatorService) {
+                return new ListItem(data, userProfileService, moderatorService);
+            }
 
-            private ListItem(BannedUserProfileData bannedUserProfileData) {
+            private ListItem(BannedUserProfileData bannedUserProfileData,
+                             UserProfileService userProfileService,
+                             ModeratorService moderatorService) {
                 this.bannedUserProfileData = bannedUserProfileData;
 
                 userProfile = bannedUserProfileData.getUserProfile();
                 userName = userProfile.getUserName();
+
+                String accusedUserProfileId = userProfile.getId();
+                BannedUserModeratorData moderatorData = moderatorService.getPersistableStore()
+                        .getBannedUserModeratorDataMap()
+                        .get(accusedUserProfileId);
+
+                if (moderatorData != null) {
+                    reporterUserProfileId = moderatorData.getReporterUserProfileId();
+                    banReason = StringUtils.toOptional(moderatorData.getBanReason())
+                            .orElse(Res.get("data.na"));
+
+                    if (reporterUserProfileId != null && !reporterUserProfileId.isEmpty()) {
+                        reporterUserProfile = userProfileService.findUserProfile(reporterUserProfileId);
+                        reporterUserName = reporterUserProfile.map(UserProfile::getUserName)
+                                .orElse(Res.get("data.na"));
+                    } else {
+                        reporterUserProfile = Optional.empty();
+                        reporterUserName = Res.get("data.na");
+                    }
+                } else {
+                    reporterUserProfileId = null;
```

Comment:
Should be Res.get("data.na"), but as its not used outside the ListItem, it should be converted to a local variable.